src/plugins/dashboard/public/application/top_nav/editor_menu.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiContextMenu, EuiContextMenuPanelItemDescriptor } from '@elastic/eui';
+import { METRIC_TYPE } from '@kbn/analytics';
+import { i18n } from '@kbn/i18n';
+import { IconType } from '@elastic/eui/src/components/icon/icon';
+import { EuiContextMenuItemIcon } from '@elastic/eui/src/components/context_menu/context_menu_item';
+import { SolutionToolbarPopover } from '../../../../presentation_util/public';
+import { EmbeddableFactoryDefinition, EmbeddableInput } from '../../services/embeddable';
+import { BaseVisType, VisGroups, VisTypeAlias } from '../../../../visualizations/public';
+import { useKibana } from '../../services/kibana_react';
+import { DashboardAppServices } from '../types';
+import { DashboardContainer } from '..';
+import { DashboardConstants } from '../../dashboard_constants';
+
+interface Props {
+  /** Dashboard container */
+  dashboardContainer: DashboardContainer;
+  /** Handler for creating new visualization of a specified type */
+  createNewVisType: (visType: BaseVisType | VisTypeAlias) => () => void;
+}
+
+interface FactoryGroup {
+  id: string;
+  appName: string;
+  icon: IconType;
+  panelId: number;
+  factories: EmbeddableFactoryDefinition[];
+}
+
+export const EditorMenu = ({ dashboardContainer, createNewVisType }: Props) => {
+  const {
+    embeddable,
+    visualizations,
+    usageCollection,
+  } = useKibana<DashboardAppServices>().services;
+
+  const trackUiMetric = usageCollection?.reportUiCounter.bind(
+    usageCollection,
+    DashboardConstants.DASHBOARDS_ID
+  );
+
+  const createNewAggsBasedVis = useCallback(
+    async () =>
```

Comment:
Why async?

Reply:
It doesn't need to be async, I think it was just leftover from copying another callback. I'll fix it.